internal/pkg/rdb/rdb.go

```diff
@@ -2,40 +2,54 @@ package rdb
 
 import (
 	"errors"
-
 	"github.com/frain-dev/convoy/util"
 	"github.com/newrelic/go-agent/v3/integrations/nrredis-v9"
 	"github.com/redis/go-redis/v9"
 )
 
 // Redis is our wrapper logic to instrument redis calls
 type Redis struct {
-	dsn    string
-	client *redis.Client
+	addresses []string
+	client    redis.UniversalClient
 }
 
 // NewClient is used to create new Redis type. This type
 // encapsulates our interaction with redis and provides instrumentation with new relic.
-func NewClient(dsn string) (*Redis, error) {
-	if util.IsStringEmpty(dsn) {
-		return nil, errors.New("redis dsn cannot be empty")
+func NewClient(addresses []string) (*Redis, error) {
+	if len(addresses) == 0 {
+		return nil, errors.New("redis addresses list cannot be empty")
 	}
 
-	opts, err := redis.ParseURL(dsn)
-	if err != nil {
-		return nil, err
+	for _, dsn := range addresses {
+		if util.IsStringEmpty(dsn) {
+			return nil, errors.New("dsn cannot be empty")
+		}
 	}
 
-	client := redis.NewClient(opts)
-
-	// Add Instrumentation
-	client.AddHook(nrredis.NewHook(opts))
+	var client redis.UniversalClient
+
+	if len(addresses) == 1 {
+		opts, err := redis.ParseURL(addresses[0])
+		if err != nil {
+			return nil, err
+		}
+
+		client = redis.NewClient(opts)
+		// Add Instrumentation
+		client.AddHook(nrredis.NewHook(opts))
+	} else {
+		client = redis.NewUniversalClient(&redis.UniversalOptions{
+			Addrs: addresses,
+		})
+		// Add Instrumentation
+		client.AddHook(nrredis.NewHook(nil))
```

Comment:
@jirevwe Will this retain instrumentation for the cluster?

Reply:
It should, though it'll have to be tested to confirm it. The param is the options which is the redis dsn in parts we don't need that for a redis cluster as the API doesn't return it

Reply:
Test it, it's retained

Reply:
Make sense!